scipy/stats/_continuous_distns.py

```diff
@@ -5881,20 +5881,36 @@ def _rvs(self, df, size=None, random_state=None):
         return random_state.standard_t(df, size=size)
 
     def _pdf(self, x, df):
+        # limiting case: fallback to the normal distribution
+        # The `fit` method and the sampling module calls `_pdf` without
+        # first converting `x` and `df` into arrays. As we use masking,
+        # we need `x` and `df` to always be arrays. So, we make sure they
+        # are arrays before masking.
+        x, df = np.asarray(x), np.asarray(df)
+        mask = (df == np.inf)
+        imask = ~mask
+        res = np.empty_like(x, dtype='d')
+        res[mask] = norm._pdf(x[mask])
         #                                gamma((df+1)/2)
         # t.pdf(x, df) = ---------------------------------------------------
         #                sqrt(pi*df) * gamma(df/2) * (1+x**2/df)**((df+1)/2)
-        r = np.asarray(df*1.0)
-        Px = (np.exp(sc.gammaln((r+1)/2)-sc.gammaln(r/2))
-              / (np.sqrt(r*np.pi)*(1+(x**2)/r)**((r+1)/2)))
+        r = df[imask]*1.0
+        res[imask] = (np.exp(sc.gammaln((r+1)/2)-sc.gammaln(r/2))
+                      / (np.sqrt(r*np.pi)*(1+(x[imask]**2)/r)**((r+1)/2)))
 
-        return Px
+        return res
 
     def _logpdf(self, x, df):
-        r = df*1.0
-        lPx = (sc.gammaln((r+1)/2) - sc.gammaln(r/2)
-               - (0.5*np.log(r*np.pi) + (r+1)/2*np.log(1+(x**2)/r)))
-        return lPx
+        x, df = np.asarray(x), np.asarray(df)
+        mask = (df == np.inf)
+        imask = ~mask
+        res = np.empty_like(x, dtype='d')
+        res[mask] = norm._logpdf(x[mask])
+        r = df[imask]*1.0
+        res[imask] = (sc.gammaln((r+1)/2) - sc.gammaln(r/2)
+                      - (0.5*np.log(r*np.pi)
+                         + (r+1)/2*np.log(1+(x[imask]**2)/r)))
```

Comment:
This PR includes manual masking, `np.where`, and `_lazywhere`. Is there a reason for switching, or should it be consistent (especially with the rest of the distributions, which tend to prefer the lazy functions) as a matter of style?

Reply:
> This PR includes manual masking, `np.where`, and `_lazywhere`. Is there a reason for switching between these strategies, or should it be consistent (especially with the rest of the distributions, which tend to prefer the lazy functions) as a matter of style?

Made it consistent. I have refactored to only use `_lazywhere`. Thanks for pointing it out!